editor/src/document/document_file.rs

```diff
@@ -562,6 +564,17 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					responses.push_front(DocumentOperation::DeleteLayer { path }.into());
 				}
 			}
+			SetViewMode(mode) => {
+				self.graphene_document.update_view_mode(mode);
+				let mut mode_update_func = |s: &mut GrapheneShape| s.style.view_mode(mode);
+				GrapheneDocument::visit_all_shapes(&mut self.graphene_document.root, &mut mode_update_func);
+				responses.push_back(
+					FrontendMessage::UpdateCanvas {
+						document: self.graphene_document.render_root(),
+					}
+					.into(),
+				);
```

Comment:
This should be changed to a `RenderDocument` Message to allow for lazy evaluation